ddtrace/contrib/internal/openai/utils.py

```diff
@@ -162,6 +162,63 @@ async def _extract_token_chunk(self, chunk):
             return
 
 
+class TracedOpenAIResponseStream(TracedOpenAIStream):
+    def __iter__(self):
+        exception_raised = False
+        try:
+            for chunk in self.__wrapped__:
+                if chunk.type == "response.completed":
+                    _tag_tool_calls(self._dd_integration, self._dd_span, chunk.response)
+                    if hasattr(chunk.response, "id"):
+                        self._dd_span.set_tag("openai.response.id", chunk.response.id)
+                    if hasattr(chunk.response, "created_at"):
+                        self._dd_span.set_tag_str("openai.response.created_at", str(chunk.response.created_at))
+                    if hasattr(chunk.response, "model"):
+                        self._dd_span.set_tag("openai.response.model", chunk.response.model)
+                    self._dd_integration.record_usage(self._dd_span, chunk.response.usage)
+                    self._dd_span.finish()
```

Comment:
Finishing the span should already be handled in the `finally` block, so unnecessary here